### ISSUE TITLE
What is our “naming” code style?

### ISSUE THREAD

**Issue by balacij** [JacquesCarette/Drasil#3336]
As the title suggests, what is our “naming” code style?

Right now we seem to have a mix of long/full and abbreviations used for name references. Should we prefer one way over the other? Can we make a hybrid with type aliases? 

(the [code style](https://github.com/JacquesCarette/Drasil/wiki/Contributor's-Guide#coding-style) does touch upon this)

**Comment by JacquesCarette** [comment]
Which code? Our own Haskell code, or our generated code?

**Comment by smiths** [comment]
I think we should have a naming style for both our Haskell code and our generated code.  For the generated code it would be great if we could change the naming conventions by selecting a different generator directive.

**Comment by balacij** [comment]
This ticket was primarily about our Haskell code, but I like @smiths' idea of extending it to the generated code too!

Regarding the Haskell code, we have `makeX`, `mkX`, `aThingsFullName`, `aThngsShrtNm`, `aThingsShrtNm'`, and similar, so it would be nice to have a codestyle guideline. The apostrophes carry very little information, but sometimes we annotate with `AThngsShrtNmWoutRefs` instead (better, in my opinion).

**Comment by JacquesCarette** [comment]
I agree that the apostrophes are bad. And that using both `makeX` and `mkX` is bad.

I am a big fan of short names for **heavily used** things. Normal functions should have reasonably descriptive names. But I find names in some ecosystems (Java comes to mind) ridiculously long. Certainly the package/module where something comes from shouldn't be repeated in a name.

I kind of feel like we should tackle this a little bit piecemeal for now. At least we should fix the glaring inconsistencies!

**Comment by balacij** [comment]
Sounds good, thank you. I'll break up this ticket this weekend into anything I can remember, but otherwise, we can just remember to clean things up as we go along, as I think you're suggesting.

**Comment by smiths** [comment]
@balacij I agree that our coding conventions should be documented on the Wiki.  If we add a PR template (#2857), we could point to the Wiki as a reminder to follow the coding convention.  We shouldn't expect developers to deduce the coding convention from the existing code, especially since it isn't currently consistent.

**Comment by B-rando1** [comment]
Should I just update the [coding style](https://github.com/JacquesCarette/Drasil/wiki/Contributor's-Guide#coding-style) section of the Contributor's Guide based on @JacquesCarette's [comment](https://github.com/JacquesCarette/Drasil/issues/3336#issuecomment-1535088189)?

I looked around a bit at Haskell style guides. I didn't see any style guides worth linking here, but this [page](https://kowainik.github.io/posts/naming-conventions) has an interesting discussion of how the various prefixes, suffixes, etc. are generally used in Haskell's naming conventions. It has less to say on length and detail of names, but it might give Haskell newcomers a good intuition for reading and writing names in Haskell.
